### PR TITLE
Prescribed velocity doc

### PR DESCRIPTION
**Pull request by danieldouglas92**
Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

*Describe what you did in this PR and why you did it.*

### Before your first pull request:

* [x] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

**Comment by cmills1095** [comment]
Note to reviewers: `sec:extending-signals' is currently a dead reference, but this will be fixed in my next pull request.

File: doc/modules/changes/20220517_danieldouglas92 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+New: Add an advection field method that
```
Comment (gassmoeller) [inline doc/modules/changes/20220517_danieldouglas92:1]: I dont think this should be part of this PR. Talk to me if you need help remove some commits from this PR.

File: include/aspect/parameters.h [hunk 2]
```diff
@@ -138,6 +138,7 @@ namespace aspect
         volume_of_fluid,
         static_field,
         fem_melt_field,
+        fem_darcy_field,
```
Comment (gassmoeller) [inline include/aspect/parameters.h:141]: same here

File: include/aspect/simulator/assemblers/advection.h [hunk 3]
```diff
@@ -46,6 +46,21 @@ namespace aspect
         compute_residual(internal::Assembly::Scratch::ScratchBase<dim>  &scratch_base) const override;
     };
 
+    template <int dim>
```
Comment (gassmoeller) [inline include/aspect/simulator/assemblers/advection.h:49]: and this part as well as all the other Darcy flow related code.

**Review by gassmoeller — COMMENTED** [review]
The documentation part looks good except for minor comments. You need to remove the commits that got mixed up in this PR.

File: cookbooks/prescribed_velocity/doc/prescribed_velocity.md [hunk 4]
```diff
@@ -7,41 +7,41 @@ In cases where it is desirable to investigate the behavior of one part of the
...
 ```
 
-&nbsp;
+```{figure-md} fig:prescribed_velocity
+<img src="prescribed_velocity.*" alt="Screenshot"  width="100%"/>
 
-  [1]: #sec:extending-signals
-  [cookbooks/prescribed_velocity]: cookbooks/prescribed_velocity
-  [2]: #fig:prescribed-velocity
+Examples of flows with prescribed internal velocities, as described in Section 5.2.9
```
Comment (gassmoeller) [inline cookbooks/prescribed_velocity/doc/prescribed_velocity.md:46]: As far as I can see this figure caption references the same section it is in. That might make sense in latex where a figure can appear far from its containing section, but is not necessary in Sphinx where it is always part of this section. 
```suggestion
Examples of flows with prescribed internal velocities.
```

**Review by gassmoeller — COMMENTED** [review]
Missed one old reference. If you fix that one this is good to go.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**